public/app/plugins/datasource/prometheus/querybuilder/components/LabelParamEditor.tsx

```diff
@@ -44,15 +45,28 @@ export function LabelParamEditor({
 
 async function loadGroupByLabels(
   query: PromVisualQuery,
-  datasource: PrometheusDatasource
+  datasource: DataSourceApi
 ): Promise<Array<SelectableValue<any>>> {
-  const labels = [{ label: '__name__', op: '=', value: query.metric }, ...query.labels];
-  const expr = promQueryModeller.renderLabels(labels);
+  if (datasource instanceof PrometheusDatasource) {
+    const labels = [{ label: '__name__', op: '=', value: query.metric }, ...query.labels];
```

Comment:
Is this line the only difference here? Could we make the `if` only on this variable?